srcpkgs/systemd-boot/files/kernel.d/systemd-boot.post-install

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+#
+# Kernel hook for systemd-boot.
+#
+# Arguments passed to this script: $1 pkgname, $2 version.
+#
+PKGNAME="$1"
+VERSION="$2"
+
+. "$ROOTDIR/etc/default/systemd-boot"
+
+if [ "$SYSTEMD_BOOT_DISABLE" ]; then
+	exit 0
+fi
+
+is_installed="$(bootctl is-installed)"
+
+if [ "$is_installed" != "yes" ]; then
+	echo "systemd-boot is not installed to the ESP"
+	echo "run 'bootctl install' to install it"
+	exit 0
+fi
+
+BOOT="${BOOT:-/boot}"
+ESP="${ESP:-"$BOOT"}"
+
+BOOT="$ROOTDIR/$BOOT"
+ESP="$ROOTDIR/$ESP"
+
+ARCH="$(xbps-uhelper arch)"
+
+entries="$ESP/loader/entries"
+name="void-$ARCH-$VERSION"
+entry="$entries/$name.conf"
+loader="$ESP/loader/loader.conf"
+
+[ -d "$ESP" ] || exit 0
+
+mkdir -p "$entries"
+
+cat <<-EOF > "$entry"
+	title Void Linux $ARCH $VERSION
+	version $VERSION
+	linux vmlinuz-$VERSION
+	initrd initramfs-$VERSION.img
+EOF
+
+# check if user provided options in /etc/default/systemd-boot
+if [ -z "$CMDLINE" ]; then
+	# get UUID from fstab
+	uuid=$(awk '$2 == "/" { print $1 }' < /etc/fstab | sed 's/^UUID=//')
+	CMDLINE="root=UUID=$uuid quiet rw"
+fi
+
+echo "Created systemd-boot entry $entry with options '$CMDLINE'"
+
+printf 'options %s\n' "$CMDLINE" >> "$entry"
+
+if grep -q ^default "$loader" 2>/dev/null; then
+	# Replace existing default entry with this entry:
+	sed -i "s/default.*/default $name.conf/" "$loader"
+else
```

Comment:
do we want to always set ourselves as default?

Reply:
Maybe only set ourselves as default if the default is already void: `s/default void.*/default $name.conf/` ?

Reply:
I guess there's also the case where an old kernel is being updated. we probably don't want to update the default in that case

Reply:
probably best to ditch the "default" and just ensure that the entries are sorted properly: https://uapi-group.org/specifications/specs/boot_loader_specification/#sorting

Reply:
sorting sounds better